src/common/ModelMetadata.ts

```diff
@@ -0,0 +1,92 @@
+export class ModelMetadata {
+    constructor(public tableName: string, public attributes: any){}
+
+    public async getAttributesForScan(args: any) : Promise<any>{
+
+        let ExclusiveStartKey, FilterExpression, Limit, ExpressionAttributeValues, ExpressionAttributeNames;
+        
+        if (args["id"]){
+            ExpressionAttributeNames = {};
+            ExpressionAttributeValues = {};
+
+            FilterExpression = `#id = :hashKey`;
+            ExpressionAttributeValues[':hashKey'] = args["id"];
+            ExpressionAttributeNames[`#id`] = 'id';
+        }
+        else{
+            Limit = args["limit"]? (args["limit"] > 0)? args["limit"] : 10 : undefined;
+
+            if (args["lastEvaluatedKey"]){
+                ExclusiveStartKey = { "id": args["lastEvaluatedKey"] };
+            }
+
+            delete args['limit'];
+            delete args['lastEvaluatedKey'];
+
+            if ( Object.keys(args).length > 0){
```

Comment:
How do you decide when to add whitespacing within parenthesis?

Reply:
Well, I always decide to use white spaces, but sometimes I miss it. And I relax on this becuase I always include the linting validations as part of CI and CD.

In this exercise, there are some files without the newline char becuase the linting corrections are pending (I put noted about it in my delivered documentations) I prefer don't to delay the delivery just because the linting was pending.